java/vector/src/main/java/org/apache/arrow/vector/TypeLayout.java

```diff
@@ -225,6 +225,117 @@ public TypeLayout visit(Duration type) {
     return layout;
   }
 
+  /**
+   * Gets the number of {@link BufferLayout}s for the given <code>arrowType</code>.
+   */
+  public static int getTypeBufferCount(final ArrowType arrowType) {
```

Comment:
Some comments here would be helpful.
Something like all fixed width types have 2 buffers -- validity and data, all variable types have validity, offset and data. For complex types, the explanation is even more helpful

Reply:
Good suggestion. I have revised the code accordingly. Please take a look.